tests/MongoDB.Bson.Tests/ObjectModel/ObjectIdTests.cs

```diff
@@ -14,20 +14,76 @@
 */
 
 using System;
+using System.Globalization;
 using System.Linq;
-using MongoDB.Bson;
+using FluentAssertions;
+using MongoDB.Bson.TestHelpers;
 using Xunit;
 
 namespace MongoDB.Bson.Tests
 {
     public class ObjectIdTests
     {
+        [Theory]
+        [InlineData(0xFFFFFE, 0xFFFFFF)]
+        [InlineData(0xFFFFFF, 0)]
+        [InlineData(0x1000000, 1)]
+        [InlineData(0x1FFFFFE, 0xFFFFFF)]
+        [InlineData(0x1FFFFFF, 0)]
+        [InlineData(0x2000000, 1)]
+        public void Ensure_that_counter_is_resetted_after_max_value(int incrementValue, int expectedIncrement)
+        {
+            ObjectIdReflector.__staticIncrement(incrementValue);
+            var oid = ObjectId.GenerateNewId();
+            var resultCounter = oid.Increment;
+            resultCounter.Should().Be(expectedIncrement);
+        }
+
+        [Fact]
+        public void Ensure_that_machine_value_is_not_the_same_if_created_from_different_appdomain()
```

Comment:
Probably this test is the best I can do for the test:

    Ensure that after a new process is created through a fork() or similar process creation operation, the "random number unique to a machine and process" is no longer the same as the parent process that created the new process.

Reply:
Using an AppDomain to test this is a nice idea. Congrats!